### PR TITLE
feat: delete dependnecy button through change request

### PR DESCRIPTION
**Pull request by kwasniew**
## About the changes

Delete dependency icon can select whether to use direct API or change request API.

Screenshot 2023-10-10 at 13 00 25

Screenshot 2023-10-10 at 13 00 55

## Discussion points

A few interesting findings in this PR:
* I managed to find how to test toasts
* I managed to find how to test permissions with and without change requests

### CODE REVIEW

File: frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewSidePanel/FeatureOverviewSidePanelDetails/FeatureOverviewSidePanelDetails.test.tsx [hunk 1]
```diff
@@ -122,6 +163,46 @@ test('delete dependency', async () => {
...
     userEvent.click(deleteButton);
+
+    await screen.findByText('Dependency removed');
+});
+
+test('delete dependency with change request', async () => {
+    setupChangeRequestApi();
+    render(
+        <>
+            <ToastRenderer />
```
Comment (kwasniew) [inline frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewSidePanel/FeatureOverviewSidePanelDetails/FeatureOverviewSidePanelDetails.test.tsx:174]: this is how we can test toasts

File: frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewSidePanel/FeatureOverviewSidePanelDetails/FeatureOverviewSidePanelDetails.test.tsx [hunk 2]
```diff
@@ -122,6 +163,46 @@ test('delete dependency', async () => {
...
+                        dependencies: [{ feature: 'some_parent' }],
+                        children: [] as string[],
+                    } as IFeatureToggle
+                }
+                header={''}
+            />
+        </>,
+        {
+            permissions: [
+                /* deliberately no permissions */
```
Comment (kwasniew) [inline frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewSidePanel/FeatureOverviewSidePanelDetails/FeatureOverviewSidePanelDetails.test.tsx:189]: this is how we can test change request permissions

File: frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewSidePanel/FeatureOverviewSidePanelDetails/FeatureOverviewSidePanelDetails.test.tsx [hunk 3]
```diff
@@ -122,6 +163,46 @@ test('delete dependency', async () => {
...
+
+    const actionsButton = await screen.findByRole('button', {
+        name: /Dependency actions/i,
+    });
+    userEvent.click(actionsButton);
+
+    const deleteButton = await screen.findByText('Delete');
+    userEvent.click(deleteButton);
+
+    await screen.findByText('Change added to a draft');
```
Comment (kwasniew) [inline frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewSidePanel/FeatureOverviewSidePanelDetails/FeatureOverviewSidePanelDetails.test.tsx:205]: this is from a toast

File: frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewSidePanel/FeatureOverviewSidePanelDetails/DependencyRow.tsx [hunk 4]
```diff
@@ -9,11 +9,67 @@ import { useFeature } from 'hooks/api/getters/useFeature/useFeature';
...
+    useHasProjectEnvironmentAccess,
+} from 'hooks/useHasAccess';
+import { useChangeRequestApi } from '../../../../../../hooks/api/actions/useChangeRequestApi/useChangeRequestApi';
+import { usePendingChangeRequests } from '../../../../../../hooks/api/getters/usePendingChangeRequests/usePendingChangeRequests';
+import useToast from '../../../../../../hooks/useToast';
+import { useHighestPermissionChangeRequestEnvironment } from '../../../../../../hooks/useHighestPermissionChangeRequestEnvironment';
+import { useChangeRequestsEnabled } from '../../../../../../hooks/useChangeRequestsEnabled';
+import { formatUnknownError } from '../../../../../../utils/formatUnknownError';
+
+const useDeleteDependency = (project: string, featureId: string) => {
```
Comment (kwasniew) [inline frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewSidePanel/FeatureOverviewSidePanelDetails/DependencyRow.tsx:20]: delete with or without change request

File: frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewSidePanel/FeatureOverviewSidePanelDetails/FeatureOverviewSidePanelDetails.test.tsx (hunk 3 above)
Comment (kwasniew) [reply frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewSidePanel/FeatureOverviewSidePanelDetails/FeatureOverviewSidePanelDetails.test.tsx:205]: if we can get a toast it means that API call that we set up worked fine

File: frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewSidePanel/FeatureOverviewSidePanelDetails/FeatureOverviewSidePanelDetails.test.tsx (hunk 2 above)
Comment (kwasniew) [reply frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewSidePanel/FeatureOverviewSidePanelDetails/FeatureOverviewSidePanelDetails.test.tsx:189]: I deliberately omit the permissions and we should still be able to add things to change request

**Review by FredrikOseberg — APPROVED** [review]
Nice. Looks good.

**Pull request merged**